rotkehlchen/api/rest.py

```diff
@@ -2297,6 +2298,73 @@ def get_asset_icon(
 
         return response
 
+    @staticmethod
+    def _get_current_assets_price(assets: List[Asset]) -> Dict[str, Any]:
+        """Return the current USD price of the assets.
+        """
+        log.info(
+            f'Querying the current prices for: {", ".join(asset.identifier for asset in assets)}',
+        )
+        assets_price = {asset: Inquirer().find_usd_price(asset) for asset in assets}
+        return _wrap_in_ok_result(process_result(assets_price))
+
+    def get_current_assets_price(
+            self,
+            assets: List[Asset],
+            async_query: bool,
+    ) -> Response:
+        if async_query:
+            return self._query_async(
+                command='_get_current_assets_price',
+                assets=assets,
+            )
+
+        response = self._get_current_assets_price(assets)
+        return api_response(_wrap_in_ok_result(response['result']), status_code=HTTPStatus.OK)
+
+    def _get_historical_assets_price(
+            self,
+            assets_timestamp: List[Tuple[Asset, Timestamp]],
+    ) -> Dict[str, Any]:
+        """Return the USD price of the assets at the given timestamps.
+
+        NB: the current implementation only returns one price per asset. Given
```

Comment:
That's not good. If multiple combos of the same asset with different timestamps are given then all of them should be returned. Not only the last timestamp.

Reply:
I specifically asked about this on discord and the general response was we don't care at the moment. I asked as it really changes the response. This is why I put the comment :S. For me it make sense to return them all so happy to change it.